ng2-material/components/menu/menu.ts

```diff
@@ -0,0 +1,125 @@
+import {Component, ViewEncapsulation, ElementRef, Renderer} from "angular2/core";
+import {DOM} from "angular2/src/platform/dom/dom_adapter";
+
+
+@Component({
+  selector: 'md-menu',
+  template: `<ng-content></ng-content>`,  
+  encapsulation: ViewEncapsulation.None
+})
+export class MdMenu {
+  private _menuElement: HTMLElement;
+  private _triggerElement: Element;
+  private _menuContent: Element;
+  private _menuContainers: any;
+  private _isOpen: boolean = false;
+
+  constructor(private _element: ElementRef, private _renderer: Renderer) {
+    this._menuElement = this._element.nativeElement;
+  }
+
+  ngAfterContentInit() {
+    if (this._menuElement.children.length != 2) {
+      throw Error('Expected two children elements.');
+    }
+
+    this.initMenu();
+  }
+
+  initMenu() {
+    var body = DOM.query('body');
+    
+    this._renderer.setElementClass(this._menuElement, 'md-menu', true);
+    this._renderer.setElementClass(this._menuElement, '_md', true);
+
+    this._triggerElement = this._menuElement.firstElementChild;
```

Comment:
I set classes and styles. I create a menuContainers and append menuContent in it.
Then I append the menuContainers in body.